### PR TITLE
deserialize specific geo types

### PR DESCRIPTION
**Pull request by michaelkirk**
- [x] I agree to follow the project's [code of conduct](https://github.com/georust/geo/blob/master/CODE_OF_CONDUCT.md).
- [x] I added an entry to `CHANGES.md` if knowledge of this change could be valuable to users.
- [x] I ran cargo fmt
---

The first two commits are just no-ops to move things around. But the final commit is the interesting one. With the exciting bit at https://github.com/georust/wkt/pull/99/files#diff-95bc9d4f4b531a38268ea56c3b4db680f15a6ab9a2bed1b257ad64217f477e5bR61

The new feature is the introduction of `wkt::deserialize_wkt`

```
 let json = r#"[
   { "geometry": "POINT (3.14 42)", "name": "bob's house" },
   { "geometry": "LINESTRING (0 0, 1 1, 3.14 42)", "name": "bob's route home" }
 ]"#;

 #[derive(serde::Deserialize)]
 struct MyGeomRecord {
     #[serde(deserialize_with = "wkt::deserialize_wkt")]
     pub geometry: geo_types::Geometry<f64>,
     pub name: String,
}
```

In the above case, it's pretty much equivalent to using the existing `deserialize_geometry` method, but this new method also works for _any_ of the types that impl `TryFromWkt`, so notably you can deserialize directly to the inner Geometry members when know you're expecting a particular geometry.

```
 let json = r#"{ "geometry": "LINESTRING (0 0, 1 1, 3.14 42)", "name": "bob's route home" }"#;

 #[derive(serde::Deserialize)]
 struct MyGeomRecord {
     #[serde(deserialize_with = "wkt::deserialize_wkt")]
     pub geometry: geo_types::LineString<f64>,
     pub name: String,
}
```

### CODE REVIEW

File: src/deserialize/mod.rs [hunk 1]
```diff
@@ -9,6 +13,93 @@ use std::{
     str::FromStr,
 };
 
+#[cfg(feature = "geo-types")]
+pub mod geo_types;
+
+/// Deserializes any type which implements TryFromWkt from a WKT string.
```
Comment (urschrei) [inline src/deserialize/mod.rs:19]: What do you think about including a sentence that clarifies that this functionality is intended for working with structured data (I know this is directly implied by the use of `serde`, but that might not be obvious to the beginner or casual Rust user) as opposed to WKT strings?

**Review by urschrei — APPROVED** [review]

File: src/deserialize/mod.rs (hunk 1 above)
Comment (michaelkirk) [reply src/deserialize/mod.rs]: Good point. I've tried to rework the docs to be a bit clearer, as well as re-ordered the doc examples in https://github.com/georust/wkt/pull/99/commits/5129bb9ac336471da8e173a63110a023f94d2f38

I'll merge now, but let me know if you'd like further changes.

**Comment by michaelkirk** [comment]
bors r=urschrei

File: src/deserialize/mod.rs (hunk 1 above)
Comment (urschrei) [reply src/deserialize/mod.rs]: That was the only thing that leaped out at me, and the updated docs look good!

**Pull request merged**